crypto/txhash/hash.go

```diff
@@ -0,0 +1,57 @@
+package txhash
+
+import (
+	"crypto"
+	"fmt"
+	gohash "hash"
+)
+
+var (
+	// Hash function used for transaction hashing.
+	hash Hash = crypto.SHA256
+
+	// fmt is a function that converts a byte slice to a string.
+	fmtHash = func(bz []byte) string {
+		return fmt.Sprintf("%X", bz)
+	}
+)
+
+// Hash is an interface for transaction hashing.
+type Hash interface {
+	// New returns a new hash.Hash.
+	New() gohash.Hash
+}
+
+// Bytes is a wrapper around a byte slice that implements the fmt.Stringer.
```

Comment:
unrelated to this PR, but would it be possible to use `Bytes` for the block hash formatting too? Would be nice to ensure full compatibility with ethereum formatting